File: NapakalakiRuby/pkg/NapakalakiRuby-0.0.1/lib/monster.rb
#encoding: utf-8
#Autor: Dunia Rodríguez y Emilio Reguero

require_relative "prize.rb"
require_relative "bad_consequence.rb"
module NapakalakiGame
class Monster
  attr_reader :name, :combatLevel, :prize, :bc, :MAXTREASURES
  
  @@MAXTREASURES = 10 
  def initialize(name, level, bc, prize,lc)
    @name = name
    @combatLevel = level
    @priz = prize
    @bc = bc
    @lc = 0
  end
  
  
  def getLevelsGained
    @priz.level
  end
  
  def getTreasuresGained
    @priz.treasures
  end
  
  def getCombatLevel
    @combatLevel
  end
  
  def getCombatLevelAgainstCultistPlayer
    @combatLevel + @levelChangeAgainstCultistPlayer
  end
  
  def getBadConsequence
    @bc
  end
  

  def to_s
    "Monster: #{@name}, combatLevel: #{@combatLevel} \n #{@bc} \n #{@prize}"
  end
  
end
end

File: Napakalaki/src/NapakalakiGame/Monster.java
/*
* PDOO Practicas - Napakalaki Java
* @author Emilio Reguero y Dunia Rodríguez 
*/
package NapakalakiGame;


public class Monster {
    private String name;
    private int combatLevel;
    private Prize prize;
    private BadConsequence bc;
    private int levelChangeAgainstCultistPlayer = 0;
    
    
    
    public Monster (String name, int combatLevel, BadConsequence bc, Prize prize) {
        this.name = name;
        this.combatLevel = combatLevel;
        this.prize = prize;
        this.bc = bc;
    }
    
    public Monster (String name, int combatLevel, BadConsequence bc, Prize prize, int IC){
        this.name = name;
        this.combatLevel = combatLevel;
        this.bc = bc;
        this.prize = prize;
        this.levelChangeAgainstCultistPlayer = IC;
    }
    public String getName(){
        return name;
    }
    
    public int getCombatLevel(){
        return combatLevel;
    }
    
    public int getCombatLevelAgainstCultistPlayer(){
        return (combatLevel+this.levelChangeAgainstCultistPlayer);
    }
    public Prize getPrize(){
        return prize;
    }
    
    public BadConsequence getBadConsequence(){
        return bc;
    }
    
    public int getLevelsGained(){
        return prize.getLevel();
    }
    
    public int getTreasuresGained(){
        return prize.getTreasures();
    }
    
   @Override
   public String toString(){
       return("\nName = " + name + ", CombatLevel: " + combatLevel + 
               ".\nPrize: " + prize + "\nBadConsequence: " + bc +".\n");
   }
}
